runtime/drivers/duckdb/model_executor_sqlstore_self.go

```diff
@@ -0,0 +1,101 @@
+package duckdb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type sqlStoreToSelfInputProps struct {
+	SQL string `mapstructure:"sql"`
+	DSN string `mapstructure:"dsn"`
+}
+
+func (p *sqlStoreToSelfInputProps) Validate() error {
+	if p.SQL == "" {
+		return fmt.Errorf("missing property 'sql'")
+	}
+	if p.DSN == "" {
+		return fmt.Errorf("missing property `dsn`")
+	}
+	return nil
+}
+
+type sqlStoreToSelfExecutor struct {
+	c *connection
+}
+
+var _ drivers.ModelExecutor = &sqlStoreToSelfExecutor{}
+
+func (e *sqlStoreToSelfExecutor) Concurrency(desired int) (int, bool) {
+	if desired > 1 {
+		return 0, false
+	}
+	return 1, true
+}
+
+func (e *sqlStoreToSelfExecutor) Execute(ctx context.Context, opts *drivers.ModelExecuteOptions) (*drivers.ModelResult, error) {
+	inputProps := &sqlStoreToSelfInputProps{}
+	if err := mapstructure.WeakDecode(opts.InputProperties, inputProps); err != nil {
+		return nil, fmt.Errorf("failed to parse input properties: %w", err)
+	}
+	if err := inputProps.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid input properties: %w", err)
+	}
+
+	outputProps := &ModelOutputProperties{}
+	if err := mapstructure.WeakDecode(opts.OutputProperties, outputProps); err != nil {
+		return nil, fmt.Errorf("failed to parse output properties: %w", err)
+	}
+	if err := outputProps.Validate(opts); err != nil {
+		return nil, fmt.Errorf("invalid output properties: %w", err)
+	}
+
+	// Build the model executor options with updated input and output properties
+	clone := *opts
+
+	newInputProps, err := e.modelInputProperties(opts.ModelName, opts.InputHandle.Driver(), inputProps)
+	if err != nil {
+		return nil, err
+	}
+	clone.InputProperties = newInputProps
+
+	newOutputProps := make(map[string]any)
+	err = mapstructure.WeakDecode(outputProps, &newOutputProps)
+	if err != nil {
+		return nil, err
+	}
+	clone.OutputProperties = newOutputProps
```

Comment:
Is this needed? Could we just let `selfToSelfExecutor` handle the output props (since we aren't rewriting it here)?